### PR TITLE
Documentation and tests for <seealso> when cref omits arguments 

### PR DESCRIPTION
**Pull request by davedawkins**
As promised in #789. I remembered to format all the source files this time. Let's see if I forgot any...

### CODE REVIEW

File: src/FSharp.Formatting.ApiDocs/GenerateModel.fs [hunk 1]
```diff
@@ -1019,10 +1048,16 @@ type internal CrossReferenceResolver(root, collectionName, qualify, extensions)
                 | true, (:? FSharpEntity as entity) ->
                     let urlBaseName = getUrlBaseNameForRegisteredEntity entity
 
-                    Some
+                    // See if we find the member that was intended, otherwise default to containing entity
+                    tryGetShortMemberNameFromMemberName memberName
+                    |> Option.bind (fun shortName ->
+                        findIList (entity.MembersFunctionsAndValues) (fun mfv -> mfv.DisplayName = shortName))
```
Comment (nojaf) [inline src/FSharp.Formatting.ApiDocs/GenerateModel.fs:1054]: `findIList` can be replaced with:
```fsharp
entity.MembersFunctionsAndValues |> Seq.tryFind (fun mfv -> mfv.DisplayName = shortName)
```

File: src/FSharp.Formatting.ApiDocs/GenerateModel.fs [hunk 2]
```diff
@@ -1019,10 +1048,16 @@ type internal CrossReferenceResolver(root, collectionName, qualify, extensions)
...
+                    |> Option.map mfvToCref
```
Comment (nojaf) [inline src/FSharp.Formatting.ApiDocs/GenerateModel.fs:1055]: A bit of personal a preference but I would go with:
```fsharp
|> function
    | Some mb -> Some(mfvToCref mb)
    | None ->
        Some
            { IsInternal = true
              ReferenceLink = internalCrossReference urlBaseName
              NiceName = getMemberName 2 entity.HasFSharpModuleSuffix memberName }
```

Just to make it explicit that you are handling both cases.
The code still compiles if you removed the `|> Option.defaultValue`, not handling the `None` case in the `function` will yield a warning/error.

**Review by nojaf — CHANGES_REQUESTED** [review]
This looks good, thank you!
Could you please remove that `findIList` function?

File: src/FSharp.Formatting.ApiDocs/GenerateModel.fs (hunk 1 above)
Comment (davedawkins) [reply src/FSharp.Formatting.ApiDocs/GenerateModel.fs:1054]: That's the spell I was looking for. Thank you

**Review by nojaf — APPROVED** [review]
Thank you again very much!
It is really cool to see you fix the gaps you encounter!
You could have complained or picked another tool altogether, yet you chose to be a part of the solution 🤘🚀

**Comment by davedawkins** [comment]
@nojaf  -- wait!! There are other tools ?:-D

Thank you for your kind words. The appreciation is very welcome.

**Pull request merged**